### PR TITLE
Microsoft management activity

### PR DESCRIPTION
**Pull request by guykeller**
## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/19591
fixes: https://github.com/demisto/etc/issues/18711
fixes: https://github.com/demisto/etc/issues/20004

## Description
An integration for Microsoft Management Activity API.

## Does it break backward compatibility?
   - [ ] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Integrations/MicrosoftManagementActivity/Pipfile [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+python-dateutil = "==2.8.1"
+pytz = "==2019.3"
+regex = "==2019.11.1"
+requests = "==2.22.0"
+six = "==1.13.0"
+tzlocal = "==2.0.0"
+urllib3 = "==1.25.7"
+PyJWT = "==1.7.1"
+PySocks = "==1.7.1"
+PyYAML = "==5.2"
```
Comment (ronykoz) [inline Integrations/MicrosoftManagementActivity/Pipfile]: we need all of those??

File: Integrations/MicrosoftManagementActivity/README.md [hunk 2]
```diff
@@ -0,0 +1,2 @@
+# TODO
+detailed description will be added here
```
Comment (ronykoz) [inline Integrations/MicrosoftManagementActivity/README.md]: Create readme once alex approves the docs

File: Integrations/MicrosoftManagementActivity/MicrosoftManagementActivity.yml [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+- display: First fetch start (minutes)
+  name: first_fetch_delta
+  defaultvalue: "1440"
+  type: 0
+  required: false
+- display: Content types to fetch
+  name: content_types_to_fetch
+  defaultvalue: ""
+  type: 16
+  required: false
```
Comment (ronykoz) [inline Integrations/MicrosoftManagementActivity/MicrosoftManagementActivity.yml]: what will happen if we don't enter anything?

File: Integrations/MicrosoftManagementActivity/MicrosoftManagementActivity.yml [hunk 4]
```diff
@@ -0,0 +1,208 @@
...
+  name: isFetch
+  defaultvalue: ""
+  type: 8
+  required: false
+- display: Incident type
+  name: incidentType
+  defaultvalue: ""
+  type: 13
+  required: false
+- display: Record types to fetch
```
Comment (ronykoz) [inline Integrations/MicrosoftManagementActivity/MicrosoftManagementActivity.yml]: Do we know which types exist?

File: Integrations/MicrosoftManagementActivity/MicrosoftManagementActivity.yml [hunk 5]
```diff
@@ -0,0 +1,208 @@
...
+  name: record_types_filter
+  defaultvalue: ""
+  type: 0
+  required: false
+  additionalinfo: A comma separated list of the record types you wish to fetch. Content
+    records with a record  type that isn't specified will not be fetched. If this
+    field is left empty, all record types will be fetched.
+- display: Workloads to fetch
```
Comment (ronykoz) [inline Integrations/MicrosoftManagementActivity/MicrosoftManagementActivity.yml]: same here

File: Integrations/MicrosoftManagementActivity/MicrosoftManagementActivity.yml [hunk 6]
```diff
@@ -0,0 +1,208 @@
...
+    outputs:
+    - contextPath: MicrosoftManagement.ContentRecord.ID
+      description: The ID of the record.
+      type: number
+    - contextPath: MicrosoftManagement.ContentRecord.CreationTime
+      description: The creation time of the record.
+      type: date
+    - contextPath: MicrosoftManagement.ContentRecord.RecordType
+      description: The type of the record.
+      type: Unknown
```
Comment (ronykoz) [inline Integrations/MicrosoftManagementActivity/MicrosoftManagementActivity.yml]: add types

File: Integrations/MicrosoftManagementActivity/MicrosoftManagementActivity.py [hunk 7]
```diff
@@ -0,0 +1,596 @@
...
+        super().__init__(base_url=f'{base_url}', headers=headers, verify=verify, proxy=proxy)
+        self.username = username
+        self.password = password
+        self.session = requests.Session()
+        self.session.headers = headers
+        self.tenant_id = None
+        self.suffix_template = "{}/activity/feed/subscriptions/{}"
+        self.tenant_id_suffix = ''
+        self.access_token = None
+        # TODO : add support for ms client
```
Comment (ronykoz) [inline Integrations/MicrosoftManagementActivity/MicrosoftManagementActivity.py]: is this still a todo?

File: Integrations/MicrosoftManagementActivity/MicrosoftManagementActivity.py [hunk 8]
```diff
@@ -0,0 +1,596 @@
...
+    Returning 'ok' indicates that the integration works like it is supposed to. Connection to the service is successful.
+
+    Args:
+        client: HelloWorld client
+
+    Returns:
+        'ok' if test passed, anything else will fail the test.
+    """
+
+    fetch_delta = demisto.params().get("first_fetch_delta", "1440")
```
Comment (ronykoz) [inline Integrations/MicrosoftManagementActivity/MicrosoftManagementActivity.py]: this should not be the format

File: Integrations/MicrosoftManagementActivity/MicrosoftManagementActivity.py [hunk 9]
```diff
@@ -0,0 +1,596 @@
...
+    if not fetch_delta.isdigit():
+        return "Error: first fetch start must be a positive integer."
+    fetch_delta = int(fetch_delta)
+    if fetch_delta > 1440:
+        return "Error: first fetch start cannot be more than 1440 minutes."
```
Comment (ronykoz) [inline Integrations/MicrosoftManagementActivity/MicrosoftManagementActivity.py]: let's make sure it's documented

File: Integrations/MicrosoftManagementActivity/Pipfile (hunk 1 above)
Comment (guykeller) [reply Integrations/MicrosoftManagementActivity/Pipfile:36]: Nope, looks different on my env

File: Integrations/MicrosoftManagementActivity/README.md [hunk 10]
```diff
@@ -0,0 +1,17 @@
+Microsoft Management Activity API (O365/Azure Events) should be used to retrieve content records from the various Microsoft Management Activity content types.
```
Comment (ronykoz) [inline Integrations/MicrosoftManagementActivity/README.md:1]: Please update the readme to include the docs for the integration itself too once Alex will complete is review

**Comment by kirbles19** [comment]
@guykeller Done.

**Comment by guykeller** [comment]
Closing since a more updated PR exists,

**Pull request closed without merging**